src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3190,9 +3191,29 @@ internal EffectiveParameters(ImmutableArray<TypeWithAnnotations> types, Immutabl
 
             // 'None' argument is allowed to match 'In' parameter and should behave like 'None' for the purpose of overload resolution
             // unless this is a method group conversion where 'In' must match 'In'
-            if (!isMethodGroupConversion && argRefKind == RefKind.None && paramRefKind == RefKind.In)
+            // There are even more relaxations with 'ref readonly' parameters feature:
+            // - 'ref' argument is allowed to match 'in' parameter,
+            // - 'ref', 'in', none argument is allowed to match 'ref readonly' parameter.
+            if (!isMethodGroupConversion)
             {
-                return RefKind.None;
+                if (paramRefKind == RefKind.In)
+                {
+                    if (argRefKind == RefKind.None)
+                    {
+                        return RefKind.None;
+                    }
+
+                    if (argRefKind == RefKind.Ref && ((CSharpParseOptions)argument.SyntaxTree.Options).IsFeatureEnabled(MessageID.IDS_FeatureRefReadonlyParameters))
+                    {
+                        return RefKind.Ref;
+                    }
+                }
+
+                // PROTOTYPE: Should this relaxation be also applied when `isMethodGroupConversion == true`?
```

Comment:
>// PROTOTYPE: Should this relaxation be also applied when `isMethodGroupConversion == true`? [](http://example.com/codeflow?start=16&length=92)

This is what the speclet says:
For the purpose of delegate/lambda/method group conversions [§10.7, §10.8], ref readonly can be interchanged with in modifier, but that results in a warning.

Reply:
Yes, method conversions are part of a follow up PR, this prototype comment is resolved there.